src/components/member/PasswordSetting.js

```diff
@@ -0,0 +1,228 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { toast } from 'react-toastify';
+import PropTypes from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import { Button, TextField } from '@material-ui/core';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { useMutation } from '../../config/queryClient';
+import {
+  CONFIRM_CHANGE_PASSWORD_BUTTON_ID,
+  CONFIRM_RESET_PASSWORD_BUTTON_ID,
+  USER_CONFIRM_PASSWORD_INPUT_ID,
+  USER_CURRENT_PASSWORD_INPUT_ID,
+  USER_NEW_PASSWORD_INPUT_ID,
+} from '../../config/selectors';
+import {
+  newPasswordValidator,
+  passwordValidator,
+  strengthValidator,
+} from '../../utils/validation';
+
+const useStyles = makeStyles(() => ({
+  mainContainer: {
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    margin: '12px 0px',
+  },
+  changePasswordContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    margin: '12px 0px',
+  },
+  buttonItem: {
+    display: 'flex',
+    padding: '12px',
+    flexDirection: 'column',
+    justifyContent: 'flex-end',
+    alignItems: 'flex-start',
+  },
+  note: {
+    fontSize: '12px',
+    margin: '4px 0 2px',
+  },
+}));
+
+const PasswordSetting = ({ user }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [currentPasswordError, setCurrentPasswordError] = useState(null);
+  const [newPasswordError, setNewPasswordError] = useState(null);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(null);
+  const { mutate: onUpdatePassword } = useMutation(
+    MUTATION_KEYS.UPDATE_PASSWORD,
+  );
+
+  const userId = user.get('id');
+
+  const onClose = () => {
+    setCurrentPassword('');
+    setNewPassword('');
+    setConfirmPassword('');
+  };
+
+  const handleChangePassword = () => {
```

Comment:
This function is overall difficult to understand because it lacks comments, and because you have lots of nested if clauses. I think you might take advantage of throwing errors (rather than returning strings in your validator functions), then you could try-catch the errors and set the error/notify. Returning more often would help as well (even if it's fake data, or just a boolean defining if yes or not you could change the password).

Reply:
I decided to return 'error strings' because that is the current approach used in graasp-auth. It is useful because TextField helperText prop shows a string whenever an error appears. Therefore, I use the error as the corresponding string. 
I did some refactoring to this function so it becomes clearer